### PR TITLE
ECIP-1083: Remove Contract Size Limit

### PR DESCRIPTION
**Pull request by BelfordZ**
ref: https://github.com/ethereum/EIPs/issues/1662 https://github.com/ethereum/EIPs/issues/170

### CODE REVIEW

File: _specs/ecip-draft_remove_170 [hunk 1]
```diff
@@ -0,0 +1,37 @@
+---
+ecip: <ECIP number>
+title: Remove Contract Size Limit
+author: Zachary Belford <@BelfordZ>, Mudit Gupta <@maxsam4>
+discussions-to: TBD
```
Comment (BelfordZ) [inline _specs/ecip-draft_remove_170]: Link back  to original issue for discussion for now
```suggestion
https://github.com/ethereum/EIPs/issues/1662
```

Comment (BelfordZ) [inline _specs/ecip-draft_remove_170]:
```suggestion
discussions-to: https://github.com/ethereum/EIPs/issues/1662
```

**Comment by BelfordZ** [comment]
bump

**Comment by TokenHash** [comment]
@bobsummerwill to my understanding a WIP PR can be merged in that status as an ECIP as per ECIP-1000:

> Draft ECIPs which may be in a very early stage may be entered as WIP ECIPs, which means they are a work in progress.

**Comment by bobsummerwill** [comment]
Right you are, @TokenHash!  I was not aware of that status.
In that case, I see no reason why this should not be merged right now.

**Comment by BelfordZ** [comment]
needs 1 more editor approval, and I dont think self-approval is appropriate.

bump @meowsbits

**Comment by soc1c** [comment]
WIP shouldn't be merged as the P stands for progress. something in progress cannot be proposed. please proceed to draft stage once you want to propose something and are confident this is a good first draft.

**Comment by TokenHash** [comment]
@soc1c 

> WIP shouldn't be merged as the P stands for progress. something in progress cannot be proposed. please proceed to draft stage once you want to propose something and are confident this is a good first draft.

Makes sense. If that is the case, then I am going to adjust ECIP-0001 to show WIP can be entered as PRs, but can only advance to merged only if they are stable drafts.

**Comment by TokenHash** [comment]
@bobsummerwill @BelfordZ @soc1c I added/modified this text on ECIP-0001 to reflect the WIP rule:

> `Draft` ECIPs which may be in a very early stage may be entered as `WIP` pull requests, which means they are a work in progress. However, `WIP` ECIP pull requests may not be merged into the ECIP repository unless authors feel confident enough about their proposals and moved them to `Draft` status.

File: _specs/ecip-1083.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+---
+ecip: 1083
+title: Remove Contract Size Limit
+author: Zachary Belford <@BelfordZ>, Mudit Gupta <@maxsam4>
+discussions-to: https://github.com/ethereum/EIPs/issues/1662
+status: WIP
```
Comment (BelfordZ) [inline _specs/ecip-1083.md]:
```suggestion
status: Draft
```

**Comment by bobsummerwill** [comment]
LGTM

**Comment by YazzyYaz** [comment]
I can approve it, but first it needs a License unless you specified it somewhere.

File: _specs/ecip-1083.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+---
+ecip: 1083
+title: Remove Contract Size Limit
+author: Zachary Belford <@BelfordZ>, Mudit Gupta <@maxsam4>
+discussions-to: https://github.com/ethereum/EIPs/issues/1662
+status: Draft
+type: Standards Track (Core)
```
Comment (meowsbits) [inline _specs/ecip-1083.md]:
```suggestion
type: Standards Track
category: Core
```

**Comment by BelfordZ** [comment]
isnt the default license of apache 2.0 assumed when no license is provided?

**Comment by meowsbits** [comment]
https://ecips.ethereumclassic.org/ECIPs/ecip-1000#ecip-licensing

File: _specs/ecip-1083.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
+---
+ecip: 1083
+title: Remove Contract Size Limit
+author: Zachary Belford <@BelfordZ>, Mudit Gupta <@maxsam4>
+discussions-to: https://github.com/ethereum/EIPs/issues/1662
+status: Draft
+type: Standards Track
+category: Core
+created: 2019-12-20
```
Comment (meowsbits) [inline _specs/ecip-1083.md]:
```suggestion
created: 2019-12-20
license: Apache-2.0
```

**Comment by bobsummerwill** [comment]
"isnt the default license of apache 2.0 assumed when no license is provided?"

There is no default license.

We never added a root license.   IP protection is still TODO.

**Pull request merged**